staging/src/k8s.io/apimachinery/pkg/runtime/converter.go

```diff
@@ -706,20 +706,18 @@ func mapToUnstructured(sv, dv reflect.Value) error {
 	if dt.Kind() == reflect.Interface && dv.NumMethod() == 0 {
 		if st.Key().Kind() == reflect.String {
 			switch st.Elem().Kind() {
-			// TODO It should be possible to reuse the slice for primitive types.
-			// However, it is panicing in the following form.
-			// case reflect.String, reflect.Bool,
-			// 	reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
-			// 	reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
-			// 	sv.Set(sv)
-			// 	return nil
+			case reflect.String, reflect.Bool,
```

Comment:
The failures from tests are related:
```
    --- FAIL: TestValidateCustomResourceDefinition/embedded_metadata_defaults (0.07s)
panic: interface conversion: interface {} is map[string]string, not map[string]interface {} [recovered]
	panic: interface conversion: interface {} is map[string]string, not map[string]interface {}
```

Reply:
Yes, just found that this TODO is not applicable anymore, see comment. Should I delete the TODO?

Reply:
Yeah - you can remove it.